src/main/java/uk/gov/register/views/ItemView.java

```diff
@@ -28,16 +49,45 @@ public ItemView(HashValue sha256hex, Map<String, FieldValue> fieldValueMap, Iter
 
     @Override
     public CsvRepresentation<Map<String, FieldValue>> csvRepresentation() {
-        Iterable<String> fieldNames = Iterables.transform(fields, f -> f.fieldName);
+        final Iterable<String> fieldNames = StreamSupport.stream(fields.spliterator(), false)
+                .map(f -> f.fieldName)
+                .collect(Collectors.toList());
+
         return new CsvRepresentation<>(Item.csvSchema(fieldNames), fieldValueMap);
     }
 
+    public String itemsTo(final String mediaType, final Provider<RegisterName> registerNameProvider) {
+        final ByteArrayOutputStream outputStream;
+        final RegisterResolver registerResolver;
+        final ItemTurtleWriter entryTurtleWriter;
+        String registerInTextFormatted = StringUtils.EMPTY;
+
+        try {
+            if (ExtraMediaType.TEXT_TTL_TYPE.getSubtype().equals(mediaType)) {
+                outputStream = new ByteArrayOutputStream();
+                registerResolver = registerName -> URI.create("http://" + registerName + ".register.gov.uk");
+                entryTurtleWriter = new ItemTurtleWriter(registerNameProvider, registerResolver);
+
+                entryTurtleWriter.writeTo(this, EntryListView.class, EntryListView.class, null, ExtraMediaType.TEXT_TTL_TYPE, null, outputStream);
+
+                registerInTextFormatted = new String(outputStream.toByteArray(), StandardCharsets.UTF_8);
+            } else if (ExtraMediaType.TEXT_YAML_TYPE.getSubtype().equals(mediaType)) {
+                registerInTextFormatted = yamlObjectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(getContent());
+            } else {
+                registerInTextFormatted = jsonObjectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(getContent());
+            }
+        } catch (final IOException ex) {
+            LOGGER.error("Error processing preview request. Impossible process the register items");
+        }
+
+        return StringEscapeUtils.escapeHtml(registerInTextFormatted.isEmpty() ? registerInTextFormatted : END_OF_LINE + registerInTextFormatted);
```

Comment:
Same two questions: why \n if registerInTextFormatted is not empty (as opposed to the view taking care of this) and do we notify the user there was an error if there was an error?

Reply:
Same two answers.
1. Because the content of the <pre> element in the view needs the \n to show properly the first line due to the JS modifications prismjs does.
2. No, we just return empty. I haven't seen any mechanism of returning proper errors in the views. If there is one, can you point me to it?

Reply:
Discussed.